services/blockchain-app-registry/shared/metadata.js

```diff
@@ -352,31 +354,27 @@ const resolveTokenMetaInfo = async (tokenInfoFromDB) => {
 
 	const tokensMeta = [];
 	const processedChainIDs = new Set();
-	const resultTokenIDsSet = tokenInfoFromDB.reduce(
-		(accSet, tokenInfo) => accSet.add(tokenInfo.tokenID),
-		new Set(),
-	);
+	const resultTokenIDsSet = new Set(tokenInfoFromDB.map(tokenInfo => tokenInfo.tokenID));
 
-	await BluebirdPromise.map(
-		tokenInfoFromDB,
-		async (entry) => {
-			const chainID = entry.tokenID.substring(0, LENGTH_CHAIN_ID);
+	// eslint-disable-next-line no-restricted-syntax
+	for (const entry of tokenInfoFromDB) {
+		const chainID = entry.tokenID.substring(0, LENGTH_CHAIN_ID);
 
-			// Avoid reading same token metadata file multiple times
-			if (processedChainIDs.has(chainID)) return;
+		if (!processedChainIDs.has(chainID)) {
```

Comment:
As it is a long code block would it be more readable if we use `continue` instead of indentation? Also, would it help others to understand if we keep the comment explaining why we are skipping the indexing for this particular token?